### PR TITLE
fix: handle unicode dirnames and filenames

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

**Review by vojtechsimetka — COMMENTED** [review]
I think better would be to properly encode such characters just as we do in the `gateway-ui`. I thought this is in the bee-dashboard already but it is not. https://github.com/ethersphere/gateway-ui/pull/200/files

**Comment by Cafe137** [comment]
With this change I was able to upload this structure as a website, and the unicode file on its own too:

image

**Comment by Cafe137** [comment]
https://bah5acgzanq7s5t4dlbz5mjo7ypaync5aakxkq4444uagmjusmykgrb66rjwq.bzz.link/%E2%80%94%F0%9D%96%86%F0%9D%96%86%F0%9F%99%87/%F0%9D%96%88%F0%9D%96%88.txt

**Comment by bee-worker** [comment]
## 🐝 PR preview in Swarm

**Preview URL:** https://bah5acgzafmcqratmzop4y3ncny76fwzgbwj4rh5redaiqxdmdqlgqzuf53sq.bzz.link
**Swarm Hash:** `2b0508826ccb9fcc6da26e3fe2db260d93c89fb120c0885c6c1c16686685eee5`
**Commit Hash:** 8c538622ee7542c680345fce317bc9ac0f551699
**Commit Message:** fix: handle unicode dirnames and filenames

**Review by vojtechsimetka — APPROVED** [review]

File: src/pages/files/Upload.tsx [hunk 1]
```diff
@@ -148,7 +148,12 @@ export function Upload(): ReactElement {
...
       .catch(e => {
         console.error(e) // eslint-disable-line
-        enqueueSnackbar(`Error uploading: ${e.message}`, { variant: 'error' })
+
+        if (e.message.includes('String contains non ISO-8859-1 code point')) {
+          enqueueSnackbar('Filename contains invalid characters', { variant: 'error' })
+        } else {
+          enqueueSnackbar(`Error uploading: ${e.message}`, { variant: 'error' })
+        }
         setUploading(false)
```
Comment (vojtechsimetka) [inline src/pages/files/Upload.tsx:157]: Can you revert this? Probably will never happen now.

Comment (Cafe137) [reply src/pages/files/Upload.tsx:157]: Yes but first I am going to test this in more browsers and add more test cases. Bit paranoid with these changes. 🙂

**Pull request merged**